### ISSUE TITLE
LocalStorage for Caching

### ISSUE THREAD

**Issue by ericrange** [Spline/Aurora#13]
all intern links on a delivered website should be precached in the localstorage.
e.g. if i stay longer more than 5 seconds on a page, start prefetching all relevant information for further sites. this precache contains json data and is the client-side equivalent to the redis cache (server-side).

**Comment by pablosichert** [comment]
That's definitely worth an idea. Alternatively we could think of limiting that to prefetch when hovering over links? That's up to discussion I would say.

**Comment by ericrange** [comment]
If you precache the data whenever you hover a link, i think this is already too late. but maybe you could just precache the intern links in the main content section only.

**Comment by pablosichert** [comment]
Yeah we should make a priority order for content prefetching and maybe additionally prefetch on hover (and validate cache?).
I would limit this everything on mobile, though.

**Comment by ericrange** [comment]
why just for mobile? acutally you need a very low expiration time for this precache. maybe 5 minutes. aurora has the claim to be a real time cms. So you need precaching also for desktop browsers.

**Comment by pablosichert** [comment]
The point being made was to limit the content prefetching _on_ mobile and not _for_ mobile.

It's arguable if with we could do it anyway with today's flat rates and/or WLAN usage. I'm pretty sure there is no problem, especially if we start prefetching the JSON objects (and not images).
